boa_engine/src/vm/opcode/mod.rs

```diff
@@ -126,32 +126,44 @@ macro_rules! generate_impl {
                 unsafe { std::mem::transmute(value) }
             }
 
+            const STR_MAP: [&'static str; 256] = {
+                let mut map = [""; 256];
+                $(
+                    map[Self::$Variant as usize] = $Variant::NAME;
+                )*
+                map
+            };
+
             /// Name of this opcode.
             #[must_use]
             pub const fn as_str(self) -> &'static str {
-                match self {
-                    $(
-                        Self::$Variant => $Variant::NAME
-                    ),*
-                }
+                Self::STR_MAP[self as usize]
             }
 
+            const INSTRUCTION_STR_MAP: [&'static str; 256] = {
+                let mut map = [""; 256];
+                $(
+                    map[Self::$Variant as usize] = $Variant::INSTRUCTION;
+                )*
+                map
+            };
+
             /// Name of the profiler event for this opcode.
             #[must_use]
             pub const fn as_instruction_str(self) -> &'static str {
-                match self {
-                    $(
-                        Self::$Variant => $Variant::INSTRUCTION
-                    ),*
-                }
+                Self::INSTRUCTION_STR_MAP[self as usize]
             }
 
+            const EXECUTE_FN_MAP: [&dyn Fn(&mut Context) -> JsResult<ShouldExit>; 256] = {
+                let mut map: [&dyn Fn(&mut Context) -> JsResult<ShouldExit>; 256]  = [&|_| { unreachable!(); }; 256];
```

Comment:
I'm pretty sure all variants only use plain functions, so this could be better expressed with an array of function pointers:
`&[fn(&mut Context) -> JsResult<ShouldExit>]`.

Also applies the previous comment about the array expression.